apps/web/src/pages/auth/components/LoginForm.tsx

```diff
@@ -48,40 +51,59 @@ export function LoginForm({ email, invitationToken }: LoginFormProps) {
     }
   >((data) => api.post('/v1/auth/login', data));
 
-  useEffect(() => {
-    (async () => {
-      if (!tokenInQuery) {
-        return;
-      }
+  const handleLoginInUseEffect = async () => {
+    // if currentUser is true, it means user exists, then while accepting invitation, InvitationPage will handle accept this case
+    if (currentUser) {
+      return;
+    }
 
-      if (!invitationToken && (!organizationId || !environmentId)) {
-        await login(tokenInQuery, ROUTES.AUTH_APPLICATION);
+    // if token from github is not present
+    if (!tokenInQuery) {
+      return;
+    }
 
-        return;
-      }
+    // handle github login after invitation
+    if (invitationTokenFromGithub) {
+      await login(tokenInQuery);
+      const updatedToken = await acceptInvite(invitationTokenFromGithub);
 
-      if (isFromVercel) {
-        await login(tokenInQuery);
-        startVercelSetup();
+      if (updatedToken) {
+        await login(updatedToken, isRedirectedFromLoginPage === 'true' ? ROUTES.WORKFLOWS : ROUTES.AUTH_APPLICATION);
```

Comment:
Ouch. This has to be done this way as the accept invitation endpoint is [authenticated](https://github.com/novuhq/novu/blob/17230b1232f0cb19a7c697372ad7876c674cef51/apps/api/src/app/invites/invites.controller.ts#L66) but it shouldn't.

Removing the `UserGuard` is a breaking change, but I'd say the current need for an authentication token is wrong, so I feel confident making the change in a follow-up PR. 

What do you think?

Reply:
I think, the  user should exist before accepting invitation. So current `UserGuard` is correct